docs/development/development.md

```diff
@@ -0,0 +1,258 @@
+
+</details>
+
+---
+
+### **How to set up a local development environment, using the Makefile?**
+
+<details>
+<summary>Answer</summary>
+
+To set up a local development environment, it is recommended to use an editor that supports Go debugging, such as Visual Studio Code or GoLand. The Makefile includes several useful commands for the development environment:
+
+- `make tools`: Installs the necessary dependencies for the project.
+- `make format-go`: Runs Go formatting, useful if you don't have automatic formatting enabled in the editor.
+- `make license-add`: Adds a license to newly added files.
+- `make lint-go`: Runs Go linting to check for style issues or common errors.
+- `make test-unit`: Runs unit tests.
+- `make test-integration`: Runs integration tests.
+
+Other commands in the Makefile are less relevant during the development cycle, so these are the main ones to use.
+
+</details>
+
+---
+
+### **What's the release process for a new version?**
+
+<details>
+<summary>Answer</summary>
+
+The release process for a new version is documented at [this link](https://github.com/sighupio/fury-distribution/blob/main/MAINTENANCE.md#furyctl). If the release is not tied to `fury-distribution`, it's enough to create a tag and release it. However, if the release is dependent on new versions of the distribution, the process may be more complex and require updating `fury-distribution` before releasing a new version.
+
+</details>
+
+---
+
+### **How does the template engine works and what are the available features?**
+
+<details>
+<summary>Answer</summary>
+
+The template engine used is the standard Go template engine, which also leverages the `https://github.com/Masterminds/sprig` library. Sprig provides several additional functions for templates, such as string manipulations, date formatting, and other utilities not included in Go's native template engine.
+
+</details>
+
+---
+
+### **How logging is implemented, libraries used, and conventions (e.g., log messages)?**
+
+<details>
+<summary>Answer</summary>
+
+Logging in the project is implemented using the `https://github.com/sirupsen/logrus` library, which is one of the most popular logging libraries for Go. There are no strict conventions for log messages, but Logrus supports various logging levels (info, error, debug) and can output logs in different formats, including plain text and JSON.
+
+</details>
+
+---
+
+### **Are there any critical parts of the project that require special attention during future development?**
+
+<details>
+<summary>Answer</summary>
+
+There are no critical parts of the project that require immediate attention. However, simplifying the template engine and reducing code duplication would help future development by making the code easier to maintain and extend.
```

Comment:
> simplifying the template engine

what do you mean by this? how is it complex? In the QA about the template engine nothing is said in this regard.

Reply:
The template engine is fine, but the code that uses it to create the various folders where it copies the templates can be re-engineered and simplified.